mbed-client-libservice/ip_fsc.h

```diff
@@ -22,8 +22,8 @@
 #define NEXT_HEADER_UDP     0x11
 #define NEXT_HEADER_ICMP6   0x3A
 
-extern uint16_t ip_fcf_v(uint_fast8_t count, const ns_iovec_t vec[static count]);
```

Comment:
@kjbracey-arm, is the dynamic array size passed in used to help catch out of bounds errors?  If so, would it be preferable to change this to a macro (which can be defined away for IAR)?  Something like this:
```
extern uint16_t ip_fcf_v(uint_fast8_t count, const ns_iovec_t vec[array_size(count)]);
```

Let me know, what you think and if so I'll update the PR.